### PR TITLE
Make setTags method concurrency-safe

### PR DESCRIPTION
**Pull request by Magomogo**
Concurrency issues are hard to catch, but the tests that I've written fail in the exact right way at my computer.

This change is already tested in concurrent environment, works safely.

### CODE REVIEW

File: store/go_cache.go [hunk 1]
```diff
@@ -89,15 +91,21 @@ func (s *GoCacheStore) setTags(ctx context.Context, key interface{}, tags []stri
...
+		s.mu.RLock()
 		if _, exists := cacheKeys[key.(string)]; exists {
+			s.mu.RUnlock()
 			continue
 		}
+		s.mu.RUnlock()
 
 		if cacheKeys == nil {
 			cacheKeys = make(map[string]struct{})
 		}
```
Comment (eko) [inline store/go_cache.go:104]: I think you could avoid using `Lock()` and `Unlock()` each time by using a [`sync.Map`](https://pkg.go.dev/sync#Map).

What do you think?

**Review by eko — COMMENTED** [review]
Thank you for fixing this concurrency issue @Magomogo. I just proposed a little change to have an event better management.

File: store/go_cache.go (hunk 1 above)
Comment (Magomogo) [reply store/go_cache.go:104]: I very appreciate the code review, thanks!

I have not much experience with Go yet, so, correct me if I wrong. According to the docs, the sync map is designed for the cases when:

> the entry for a given key is only ever written once but read many times

So it comes with a price, but reduces lock contention for multi core processors. Do you think this is a case here?

Comment (eko) [reply store/go_cache.go:104]: There is a second case to be used:
> when multiple goroutines read, write, and overwrite entries for disjoint sets of keys

I agree with your point of view, maybe we could try to benchmark it with https://github.com/eko/gocache/blob/master/store/go_cache_bench_test.go#L13 ;-)

Comment (eko) [reply store/go_cache.go:104]: I'm back after some benchmarks, it seems that the simple map[] + mutex solution still be (a little) more efficient than the sync.Map solution here:

###  Set with map[] + mutex:
```go
BenchmarkGoCacheSet/1-16  	 2080497	       555.6 ns/op	      96 B/op	       7 allocs/op
BenchmarkGoCacheSet/2-16  	  902278	      1153 ns/op	     192 B/op	      14 allocs/op
BenchmarkGoCacheSet/4-16  	  482770	      2359 ns/op	     384 B/op	      28 allocs/op
BenchmarkGoCacheSet/8-16  	  222925	      4898 ns/op	     768 B/op	      56 allocs/op
BenchmarkGoCacheSet/16-16 	  118359	      9730 ns/op	    1536 B/op	     112 allocs/op
BenchmarkGoCacheSet/32-16 	   63066	     18599 ns/op	    3073 B/op	     224 allocs/op
BenchmarkGoCacheSet/64-16 	   31719	     36729 ns/op	    6147 B/op	     448 allocs/op
BenchmarkGoCacheSet/128-16         	   16136	     78881 ns/op	   14344 B/op	     896 allocs/op
BenchmarkGoCacheSet/256-16         	    6294	    173394 ns/op	   34837 B/op	    2560 allocs/op
BenchmarkGoCacheSet/512-16         	    3327	    371194 ns/op	   69677 B/op	    5120 allocs/op
BenchmarkGoCacheSet/1024-16        	    1606	    681796 ns/op	  147551 B/op	   10240 allocs/op
```

###  Set with a sync.Map:
```go
BenchmarkGoCacheSet/1-16  	 1900226	       608.9 ns/op	     144 B/op	       8 allocs/op
BenchmarkGoCacheSet/2-16  	  903712	      1267 ns/op	     288 B/op	      16 allocs/op
BenchmarkGoCacheSet/4-16  	  428809	      2557 ns/op	     576 B/op	      32 allocs/op
BenchmarkGoCacheSet/8-16  	  208640	      5123 ns/op	    1152 B/op	      64 allocs/op
BenchmarkGoCacheSet/16-16 	  113078	      9948 ns/op	    2305 B/op	     128 allocs/op
BenchmarkGoCacheSet/32-16 	   58567	     20127 ns/op	    4610 B/op	     256 allocs/op
BenchmarkGoCacheSet/64-16 	   29749	     40187 ns/op	    9221 B/op	     512 allocs/op
BenchmarkGoCacheSet/128-16         	   14227	     85176 ns/op	   20493 B/op	    1024 allocs/op
BenchmarkGoCacheSet/256-16         	    6052	    195346 ns/op	   47134 B/op	    2816 allocs/op
BenchmarkGoCacheSet/512-16         	    3163	    371821 ns/op	   94267 B/op	    5632 allocs/op
BenchmarkGoCacheSet/1024-16        	    1536	    772929 ns/op	  196733 B/op	   11264 allocs/op
```

That's about the same results for getting values so let's keep it as you implemented :)

Thank you for your time!

**Review by eko — APPROVED** [review]
Thank you again @Magomogo!

File: store/go_cache.go (hunk 1 above)
Comment (Magomogo) [reply store/go_cache.go:104]: Thanks for measuring!

**Pull request merged**